R/importConservation.R

```diff
@@ -0,0 +1,164 @@
+#' pairs.
+#'
+#' \code{.importConservation} imports conservation scores from a UCSC chromosome
+#' conservation score file. Generate a big matrix (col size 2, row size equal to
+#' the number of rows of the input file) containing conservation score and
+#' chromosome position pairs.
+#'
+#' @param fName The path to the UCSC chromosome conservation score file.
+#' @param outName The path to an output file for the rCNA RDS object.
+#' @param scoringType The type of the conservation score.
+#' @param silent Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @param writeLog  Return rCNA object of COSMIC CNA data as RDS compressed data frame file.
+
+.importConservation <- function(fName,
+                                outName,
+                                scoringType,
+                                silent = FALSE,
+                                writeLog = TRUE) {
+
+    # ==== PARAMETER TYPE VALIDATION ==========================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fName,     like = "FILE_E",   checkSize = TRUE))
+    cR <- c(cR, .checkArgs(outName,     like = "a",        checkSize = TRUE))
+    cR <- c(cR, .checkArgs(scoringType,     like = "a",        checkSize = TRUE))
+    cR <- c(cR, .checkArgs(silent,       like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeLog,     like = logical(1), checkSize = TRUE))
+
+    if (length(cR) > 0) {
+        stop(cR)
+    }
+
+    # ==== READ UCSC Conservation DATA ========================================
+
+    if (!silent) {
+        cat("Reading UCSC Conservation data from ", fName, " ...\n")
+    }
+
+    conservationFile <- file(fName, open = "r")
+
+    # Read header file
+    tmp <- readLines(conservationFile, n = 1)
+    header <- unlist(strsplit(tmp, " "))
+
+    chrCol <- grep("chr", header)
+    startCol <- grep("start", header)
+
+    if (length(chrCol) == 0 || length(startCol) == 0) {
+        errorMessage <- "Invalid header."
+        stop(errorMessage)
+    }
+
+    chromosomeSource <- unlist(strsplit(header[chrCol], "="))[2]
+    currPos <- as.numeric(unlist(strsplit(header[startCol], "="))[2])
+    currIndex <- 1
+
+
+    # Determine the number of lines
+    lineCountFile <- file(fName, open = "r")
+
+    numLines <- 0
+
+    while ((numLinesRead <- length(readLines(lineCountFile, n = 200000))) > 0) {
+        numLines <- numLines + numLinesRead
+    }
+
+    # Check if numLilnes is equal to
+    if (numLines == 0) {
+        errorMessage <- "Empty file."
+        stop(errorMessage)
+    }
+
+    close(lineCountFile)
+
+
+    # Create really big matrix
+    chromosomeScores <- matrix(data = 0.0, ncol = 2, nrow = numLines)
+
+    while (length(entry <- readLines(conservationFile, n = 1)) > 0) {
+        if (!silent) {
+            .pBar(currIndex, numLines)
+        }
+        # Read a header line
+        if (nchar(entry) > 10) {
```

Comment:
I just wrote "10" - but is it safe? Also a "magic" number like that requires a comment.